backend/src/main/kotlin/app/ehrenamtskarte/backend/userdata/webservice/UserImportHandler.kt

```diff
@@ -0,0 +1,111 @@
+
+            BufferedReader(InputStreamReader(file.content())).use { reader ->
+                getCSVParser(reader).use { csvParser ->
+                    importData(csvParser)
+                }
+            }
+            ctx.status(200).result("Import successfully completed")
+        } catch (exception: UserImportException) {
+            ctx.status(400).result(exception.message ?: "Invalid input provided")
+        } catch (exception: Exception) {
+            logger.error("Failed to perform user import", exception)
+            ctx.status(500).result("Internal error occurred")
+        }
+    }
+
+    private fun getCSVParser(reader: BufferedReader): CSVParser {
+        return CSVParser(
+            reader,
+            CSVFormat.DEFAULT.builder()
+                .setHeader()
+                .setTrim(true)
+                .setSkipHeaderRecord(true)
+                .build()
+        )
+    }
+
+    private fun importData(csvParser: CSVParser) {
+        val headers = csvParser.headerMap.keys
+        val requiredColumns = setOf("userHash", "startDate", "endDate", "revoked")
+        if (!headers.containsAll(requiredColumns)) {
+            throw UserImportException("Missing required columns: ${requiredColumns - headers}")
+        }
+
+        transaction {
+            // TODO replace with koblenz.pass.app after #1428
+            val project = ProjectEntity.find { Projects.project eq "showcase.entitlementcard.app" }.single()
+
+            for (entry in csvParser) {
+                if (entry.toMap().size != csvParser.headerMap.size) {
+                    throw UserImportException("The entry [${entry.joinToString(",")}] has missing data")
+                }
+
+                val userHash = entry.get("userHash").toByteArray()
+                if (!isValidUserHash(userHash)) {
+                    throw UserImportException("Invalid userHash format: ${userHash.joinToString(",")}")
+                }
+
+                val startDate = parseDate(entry.get("startDate"))
+                val endDate = parseDate(entry.get("endDate"))
+                if (startDate.isAfter(endDate)) {
+                    throw UserImportException("Start date [$startDate] cannot be after end date [$endDate]")
+                }
+
+                val revoked = entry.get("revoked").toBoolean()
+
+                UserEntitlementsRepository.insertOrUpdateUserData(
+                    userHash,
+                    startDate,
+                    endDate,
+                    revoked,
+                    project.id.value
+                )
+            }
+        }
+    }
+
+    private fun isValidUserHash(userHash: ByteArray): Boolean {
+        // TODO implement userHash validation after #1433
+        return true
+    }
+
+    private fun parseDate(dateString: String): Instant {
+        // TODO use timezone from Koblenz config after #1428
+        try {
+            return dateStringToStartOfDayInstant(dateString, "dd.MM.yyyy", ZoneId.systemDefault())
```

Comment:
Hm do we really need a timezone. i think the system default UTC should be fine?

Reply:
true. no idea why I thought that